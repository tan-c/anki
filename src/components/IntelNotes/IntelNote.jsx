
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { IntelNotePage } from './Page';

import SearchModal from '../_layout/Modal/Search';
import FileModal from '../_layout/Modal/File';

import './intelnote.scss';
// The Main component renders one of the three provided
// Routes (provided that one matches). Both the /roster
// and /schedule routes will match any pathname that starts
// with /roster or /schedule. The / route will only match

export class MainIntelNote extends React.Component {
  render() {
    const {
      showModal,
    } = this.props;

    return (
      <main id="intelnote" data-role="main-intelnote">
        {showModal === 'search' && <SearchModal />}
        {showModal === 'file' && <FileModal />}

        <IntelNotePage />
      </main>
    );
  }
}

MainIntelNote.defaultProps = {
  showModal: '',
};

MainIntelNote.propTypes = {
  showModal: PropTypes.string,
};

function mapStateToProps(state, ownProps) {
  return {
    showModal: state.ui.getIn(['himalayan', 'showModal']),
  };
}

function mapDispatchToProps(dispatch) {
  return {
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MainIntelNote);
