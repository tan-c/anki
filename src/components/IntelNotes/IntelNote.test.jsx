import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { MainIntelNote } from './IntelNote';

function setup() {
  const props = {
  };

  return shallow(<MainIntelNote {...props} />);
}

describe('MainIntelNote Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      // expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("main[data-role='main-intelnote']")).toHaveLength(1);
    });
  });
});
