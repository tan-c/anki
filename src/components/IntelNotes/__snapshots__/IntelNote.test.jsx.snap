// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainIntelNote Unit Tests when normally set up renders correctly 1`] = `
<main
  data-role="main-intelnote"
  id="intelnote"
>
  <IntelNotePage />
</main>
`;
